shark_turbine/kernel/wave/minimize_global_loads.py

```diff
@@ -0,0 +1,237 @@
+from ..wave.constraints import (
+    Constraint,
+    HardwareConstraint,
+    WorkgroupConstraint,
+    TilingConstraint,
+)
+from .._support.tracing import CapturedTrace
+from .._support.indexing import IndexingContext, IndexSequence, IndexSymbol, IndexExpr
+from ..ops.wave_ops import Read, Write, Output, get_custom
+from ..lang.global_symbols import *
+from .utils import delinearize_index, remove_unused_operators
+from math import prod
+import torch.fx as fx
+from collections import defaultdict
+
+
+def has_write_shared_user(node: Read) -> bool:
+    idxc = IndexingContext.current()
+    return any(
+        isinstance(user, Write)
+        and user.type.address_space.subs(idxc.subs) == SHARED_ADDRESS_SPACE
```

Comment:
Should we also check that it's **only** users are write to shared memory? This will be helpful in not breaking the case where we also have computation that doesn't  shared memory but rely on the same expansion.

EDIT: I think should be OK actually, since we are not actively removing/replacing the original global reads, and have auto-remove of unused ops.